test/node-api/test_threadsafe_function/test.js

```diff
@@ -211,6 +211,15 @@ new Promise(function testWithoutJSMarshaller(resolve) {
 }))
 .then((result) => assert.strictEqual(result.indexOf(0), -1))
 
+// Make sure that threadsafe function isn't stalled when we hit
+// `kMaxIterationCount` in `src/node_api.cc`
+.then(() => testWithJSMarshaller({
+  threadStarter: 'StartThreadNonblocking',
+  maxQueueSize: binding.ARRAY_LENGTH >>> 1,
```

Comment:
Looks stupid, but there is an assert in the binding that the queue block at least once, and I didn't want to change .c code 😂